ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java

```diff
@@ -177,10 +180,75 @@ public void idempotentOperationsTest()
     assertFalse("ByteBufferReadableStreamChannel is not closed", byteBufferReadableStreamChannel.isOpen());
   }
 
+  /**
+   * Tests construction of {@link ByteBufferReadableStreamChannel} on some corner cases.
+   * <p/>
+   * Corner case list:
+   * 1. Create with a null {@link ByteBuffer}.
+   * 2. Create with a null list of {@link ByteBuffer}.
+   * 3. Create with an empty {@link ByteBuffer} list.
+   * 4. Create with a list of {@link ByteBuffer} containing a null {@link ByteBuffer}.
+   */
+  @Test
+  public void cornerCasesTest() {
+    // null byteBuffer
+    ByteBufferReadableStreamChannel readableStreamChannel;
+    try {
+      ByteBuffer content = null;
+      readableStreamChannel = new ByteBufferReadableStreamChannel(content);
+    } catch (RuntimeException e) {
+      Exception exception = (Exception) Utils.getRootCause(e);
+      assertTrue("Exception is not NullPointerException", exception instanceof NullPointerException);
+    }
+
+    // null bufferList
+    List<ByteBuffer> bufferList;
+    try {
```

Comment:
there is a lot of repeated code in the `try..catch` blocks here. Please try not to repeat code if possible. For example, this can possibly be made into a function that is something like 
```
 private void constructionFailureTest(List<ByteBuffer> bufferList, Class expectedExceptionType) {
    try {
      new ByteBufferReadableStreamChannel(bufferList);
      fail("Operations should have failed");
    } catch (RuntimeException e) {
      assertEquals("Unexpected RuntimeException", expectedExceptionType, e.getClass());
    }
  }
```
You don't need the root causing either I think.